src/tests/Common/Coreclr.TestWrapper/MobileAppHandler.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.IO;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CoreclrTestLib
+{
+    public class MobileAppHandler
+    {
+        public void InstallMobileApp(string platform, string category, string testBinaryBase, string reportBase)
+        {
+            HandleMobileApp("install", platform, category, testBinaryBase, reportBase);
+        }
+
+        public void UninstallMobileApp(string platform, string category, string testBinaryBase, string reportBase)
+        {
+            HandleMobileApp("uninstall", platform, category, testBinaryBase, reportBase);
+        }
+
+        private static void HandleMobileApp(string action, string platform, string category, string testBinaryBase, string reportBase)
+        {
+            //install or uninstall mobile app
+            string outputFile = System.IO.Path.Combine(reportBase, action, $"{category}_{action}.output.txt");
+            string errorFile = System.IO.Path.Combine(reportBase, action, $"{category}_{action}.error.txt");
+            string dotnetCmd_raw = System.Environment.GetEnvironmentVariable("__TestDotNetCmd");
+            string dotnetCmd;
+            string xharnessCmd_raw = System.Environment.GetEnvironmentVariable("XHARNESS_CLI_PATH");
+            string xharnessCmd;
+            string cmdStr;
+            string appExtension;
+            int timeout = 240000; // Set timeout to 4 mins, because the installation on Android arm64/32 devices could take up to 4 mins on CI
+
+            if(String.IsNullOrEmpty(dotnetCmd_raw))
+            {
+                dotnetCmd = "dotnet";
+            }
+            else
+            {
+                dotnetCmd = dotnetCmd_raw;
+            }
+
+            if(String.IsNullOrEmpty(xharnessCmd_raw))
+            {
+                xharnessCmd = "xharness";
+            }
+            else
+            {
+                xharnessCmd = $"exec {xharnessCmd_raw}";
+            }
+
+            if(platform == "android")
+            {
+                appExtension = "apk";
+            }
+            else
+            {
+                appExtension = "ipa";
```

Comment:
I think we actually have `.app` for Apple app bundles

Reply:
However the XHarness command below (:61) won't work for iOS anyway as it needs different arguments a bit.

Should you maybe just omit iOS now and do Android and we can see later what iOS needs?

Reply:
For this PR, we could leave the command the same as Android, if the command is not finalized yet. But our goal is to share this framework between Android and IOS. I will update the app extension.